src/server.ts

```diff
@@ -20,7 +21,27 @@ import {
 } from "./pipeline.js";
 import { getRuntimeConfig } from "./runtimeConfig.js";
 import { indexNewTorrents, TorrentLocator } from "./torrent.js";
-import { sanitizeInfoHash } from "./utils.js";
+import { formatAsList, sanitizeInfoHash } from "./utils.js";
+
+const ANNOUNCE_SCHEMA = z
+	.object({
+		guid: z.string(),
+		name: z.string(),
+		link: z.string().url(),
+		tracker: z.string(),
+	})
+	.strict();
+
+const WEBHOOK_SCHEMA = z
+	.object({
+		infoHash: z.string().length(40).nullish(),
+		path: z
+			.string()
+			.nullish()
+			.refine((path) => !path || existsSync(path)),
+	})
```

Comment:
`nullish()` is not the right primitive to use, as we shouldn't be, and don't currently, accept `null` as a value.

usage of  [`.optional()`](https://zod.dev/?id=optionals) might be better while also checking that there is at least one value after in a refine?

Reply:
I think `verifyPath` won't be necessary as the exist check should be enough. `verifyPath` also currently requires a directory and permission checks.

Reply:
Not sure if it makes it less readable as to what's going on, but also worth mentioning since they are all optional is [`partial()`](https://zod.dev/?id=partial)